src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx

```diff
@@ -0,0 +1,144 @@
+---
+title: Python agent on Microsoft Azure App Services and Container Apps
+metaDescription: How to install APM for Python on Microsoft Azure App Services and Container Apps.
+redirects:
+  - /docs/agents/python-agent/hosting-services/python-agent-microsoft-azure
+  - /docs/python/python-agent-on-microsoft-azure
+freshnessValidatedDate: never
+---
+
+Learn about special considerations for using Microsoft Azure Container Apps and App Services (formerly known as Web Apps) as a hosting service with New Relic's Python agent.
+
+## What you need [#compatibility]
+
+In addition to the [Compatibility and requirements for the Python agent](/docs/agents/python-agent/getting-started/compatibility-requirements-python-agent), you'll also need the following for this tutorial:
+
+* A workng installation of Python
+* The Windows Azure SDK for Python for your platform
+
+<Callout variant="important">
+  The Python agent cannot capture telemetry for Azure Functions. Instead, you will need to use the [Azure Functions monitoring integration](/docs/infrastructure/microsoft-azure-integrations/azure-integrations-list/azure-functions-monitoring-integration/).
+</Callout>
+
+## Installing the Python agent [#install_python]
+
+To add the New Relic Python agent to your Azure application, add `newrelic` to your `requirements.txt` file.
+
+Here's an example with a sample app:
+
+```python
+# app.py
+# Run with command "flask run"
+
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+    return "Hello, fellow humans!"
+
+if __name__ == '__main__':
+    app.run()
+```
+
+On the command line:
+
+```console
+foo@bar:~$ cat requirements.txt 
+newrelic
+flask
+```
+
+If you're using a Dockerfile, add `newrelic-admin run-program` as an entrypoint to your startup script.
+
+```lang-dockerfile
+COPY requirements.txt .
+COPY app.py .
+
+RUN pip install -r requirements.txt
+
+EXPOSE 5000
+
+ENTRYPOINT ["newrelic-admin", "run-program"]
+CMD ["flask", "run"]
+```
+
+Otherwise, initialize the agent by adding these lines to your app:
+
+```python
+# app.py
+# Run with command "flask run"
+
+import newrelic.agent   # ADDED
+from flask import Flask
+
+newrelic.agent.initialize()   # ADDED
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+    return "Hello, fellow humans!"
+
+if __name__ == '__main__':
+    app.run()
+```
+
+## Adding app settings in Azure [#azure_settings]
+
+After installation, you must configure your app. 
+
+### Azure App Services [#app-service-configuration-portal]
+
+1. Sign in to <DNT>**[portal.azure.com](https://portal.azure.com)**</DNT>.
```

Comment:
Rather than providing steps about how a third party UI exists today, we describe that they need to add the variables to their App settings, then link out to a doc about App settings in the Azure docs? I think about how often the New Relic platform UI changes and how quickly docs can fall out of sync when trying to document a product whose development we have no insight into.

Reply:
Let me think about this a bit tbh... It might be easier to keep what you have for now and if a problem comes up, we can change course. I'm not as familiar with Azure's docs or their doc hygiene -- as of right now, we don't have a way to detect 404'd links so we'd be relying on their diligence about embedding redirects with doc changes